mr/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -212,11 +212,28 @@ public static boolean hiveCatalog(Configuration conf, Properties props) {
       return Optional.empty();
     } else {
       String name = catalogName == null ? ICEBERG_DEFAULT_CATALOG_NAME : catalogName;
+
+      if (catalogType != null && isCustomCatalogSet(conf, catalogName)) {
+        throw new IllegalArgumentException(String.format("Provide only one of the following configs: '%s' or '%s'",
+                CatalogUtil.ICEBERG_CATALOG_TYPE, CatalogProperties.CATALOG_IMPL));
+      }
+
       return Optional.of(CatalogUtil.buildIcebergCatalog(name,
               getCatalogProperties(conf, name, catalogType), conf));
     }
   }
 
+  /**
+   * Returns true if {@link CatalogProperties#CATALOG_IMPL} is set.
+   * @param conf global hive configuration
+   * @param catalogName name of the catalog
+   * @return true if {@link CatalogProperties#CATALOG_IMPL} is set
+   */
+  private static boolean isCustomCatalogSet(Configuration conf, String catalogName) {
```

Comment:
If this is not reused, we might want to put the code where the method is called now

Reply:
Fixed